src/Microsoft.OData.Core/Json/IJsonWriterFromStreamFactory.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.OData.Json
+{
+    /// <summary>
+    /// The interface of factories that create <see cref="IJsonWriter"/> instances
+    /// that write directly to a <see cref="Stream"/> rather than a <see cref="TextWriter"/>.
+    /// </summary>
+    [CLSCompliant(false)]

```

Comment:
What's the benefit for this attribute?

Reply:
The `IJsonWriter` interface is not CLS-compliant, so any public API that exposes the `IJsonWriter` has to be explicitly marked as not being CLS compliant using that attribute. Otherwise, the code will not compile.

The reason `IJsonWriter` is not CLS compliant is because it exposes an `sbyte` argument in its public API via the `WriteValue(sbyte)` method. `sbyte` is not a CLS-compliant type.

For anyone curious about CLS compliance: the Common Language Specification defines a set of rules that are common to all .NET languages (that support CLS). Using a feature outside of this common specification means the code is not CLS-compliant. Check out [these docs](https://docs.microsoft.com/en-us/dotnet/standard/language-independence) to learn more about the topic.